tests/pytests/unit/test_loader.py

```diff
@@ -0,0 +1,93 @@
+"""
+Tests for salt.loader
+"""
+import os
+import shutil
+import sys
+
+import pytest
+import salt.loader
+import salt.loader_context
+import salt.utils.files
+from tests.support.helpers import dedent
+
+
+@pytest.fixture
+def loader_dir(tmp_path):
+    """
+    Create a simple directory with a couple modules to load and run tests
+    againt.
+    """
+    mod_content = dedent(
+        """
+    def __virtual__():
+        return True
+
+    def set_context(key, value):
+        __context__[key] = value
+
+    def get_context(key):
+        return __context__[key]
+    """
+    )
+    with salt.utils.files.fopen(os.path.join(tmp_path, "mod_a.py"), "w") as fp:
+        fp.write(mod_content)
+    with salt.utils.files.fopen(os.path.join(tmp_path, "mod_b.py"), "w") as fp:
+        fp.write(mod_content)
+    try:
+        yield tmp_path
+    finally:
+        shutil.rmtree(tmp_path)
+
+
+def test_loaders_have_uniq_context(loader_dir):
+    """
+    Loaded functions run in the LazyLoader's context.
+    """
+    opts = {"optimization_order": [0, 1, 2]}
+    loader_1 = salt.loader.LazyLoader([loader_dir], opts,)
+    loader_2 = salt.loader.LazyLoader([loader_dir], opts,)
+    loader_1._load_all()
+    assert loader_1.pack["__context__"] == {}
+    assert loader_2.pack["__context__"] == {}
+    loader_1["mod_a.set_context"]("foo", "bar")
+    assert loader_1.pack["__context__"] == {"foo": "bar"}
+    assert loader_1["mod_b.get_context"]("foo") == "bar"
+    with pytest.raises(KeyError):
+        loader_2["mod_a.get_context"]("foo")
+    assert loader_2.pack["__context__"] == {}
```

Comment:
**question**: Is this supposed to be testing any of the LoaderContext/NamedLoaderContext/etc functionality? Or is this ensuring that the behavior isn't changing from master to the new functionality? I was able to get this to pass against the current master simply by adding the following lines:

**loader.py**
```
from unittest.mock import MagicMock

LoadedMod = LoadedFunc = MagicMock()
```

**salt/loader_context.py**
```
from unittest.mock import MagicMock

LoaderContext = NamedLoaderContext = MagicMock()
```

Reply:
Great catch, this should have been failing without the loader changes. Fixed with https://github.com/saltstack/salt/pull/58853/commits/158a8683c5084bc7d7a56e4bd7e8b17d0410ac0f